### ISSUE TITLE
R CMD build not using R_BUILD_ARGS?

### ISSUE THREAD

**Issue by cdeterman** [travis-ci/travis-ci#4619]
@craigcitro, @eddelbuettel, @hadley

I have been developing an R package that has successfully built several times on Travis CI.  To save time on Travis I want to skip the generation of vignettes and the manual.  I try to get this by adding:

```
env:
  global:
    - R_BUILD_ARGS="--no-build-vignettes --no-manual --no-resave-data"
    - R_CHECK_ARGS="--no-build-vignettes --no-manual"
```

However, the R CMD build section shows:

```
The downloaded source packages are in
	‘/tmp/RtmpC9CwOs/downloaded_packages’
Building with: R CMD build --no-build-vignettes --no-manual --no-resave-data
$ R CMD build  .
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘gpuR’:
* checking DESCRIPTION meta-information ... OK
* cleaning src
* running ‘cleanup’
* installing the package to build vignettes
```

Where I can see it is still trying to build the vignettes.  I seems like the build flags aren't passed to the R CMD build command.  Did I miss something?

The most recent build attempt is [here](https://travis-ci.org/cdeterman/gpuR/builds/74232221).

**Comment by craigcitro** [comment]
aha -- so there's a bit of inconsistency here between what prints and what executes. that's my fault, sorry!

you actually want to use top-level config options in your `.travis.yml`, i.e.
```yaml
r_build_args: --no-build-vignettes --no-manual --no-resave-data
r_check_args: --no-build-vignettes --no-manual
```
[the code](https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/r.rb#L133-L134) completely ignores the environment variables. the issue is that i **print** the environment variable, which is clearly a holdover from r-travis, but then **use** the config option. i'll clean that up, and sorry about the confusion!

**Comment by cdeterman** [comment]
Thanks @craigcitro, that solved the problem.  However, now I need to figure out how to get Travis to wait for R CMD check when it takes longer than 10 minutes to `checking whether package ‘gpuR’ can be installed`.  Clearly the compile time for my package is beginning to take too long.

**Comment by craigcitro** [comment]
@cdeterman yeah, that's one a few people have hit (eg #3849), and it's not clear what the right solution is.
